src/pages/index.js

```diff
@@ -81,45 +145,110 @@ const newCardPopup = new PopupWithForm({
 });
 newCardPopup.setEventListeners();
 
-const profileEditPopup = new PopupWithForm({
-  popupSelector: "#profile-edit-modal",
-  handleFormSubmit: (data) => {
-    handleProfileFormSubmit(data);
-  },
+// const avatarEditPopup = new PopupWithForm({
+//   popupSelector: "#update-avatar-modal",
+// });
+// avatarEditPopup.setEventListeners();
+
+const deleteCardPopup = new PopupWithForm({
+  popupSelector: "#delete-card-modal",
 });
-profileEditPopup.setEventListeners();
+deleteCardPopup.setEventListeners();
 
 const userInfo = new UserInfo({
   profileTitleSelector: ".profile__title",
   profileDescriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+api
+  .getUserInfo(userInfo._title, userInfo._description)
+  .then((res) => {
+    userInfo.setUserInfo({ title: res.name, description: res.about });
+    userInfo.setAvatar(res.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const profileEditPopup = new PopupWithForm({
+  popupSelector: "#profile-edit-modal",
+  handleFormSubmit: (data) => {
+    api
+      .editUserInfo({ name: data.title, about: data.description })
+      .then((res) => {
+        userInfo.setUserInfo({ title: res.name, description: res.about });
+        profileEditPopup.renderLoading(true);
+        handleProfileFormSubmit({ title: data.name, description: data.about });
+        api.getUserInfo(userInfo._title, userInfo._description);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  },
+});
+const updateAvatar = new PopupWithForm({
+  popupSelector: "#update-avatar-modal",
+  handleFormSubmit: (avatar) => {
+    api
+      .updateAvatar({ avatar: avatar.link })
+      .then((res) => {
+        console.log(res);
+        userInfo.setAvatar(res.avatar);
+        updateAvatar.renderLoading(true);
+        // api.getAvatar({ avatar: avatar.link });
+        // userInfo.getAvatar(res.avatar);
+        // handleFormSubmit({ avatar: avatar.link });
+        updateAvatar.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  },
 });
 
 /*EVENT HANDLERS*/
 
+profileEditPopup.setEventListeners();
+
+function handleAddCardFormSubmit({ name, link }) {
+  api
+    .createNewCard(name, link)
+    .then((res) => {
+      const newCard = renderCard(res);
+      cardsWrap.addItem(newCard);
+      newCardPopup.renderLoading(true);
+      newCardPopup.close();
+      newCardPopup.reset();
```

Comment:
Needs correction: there is no need to reset the form here, since you do that in `close` method of the PopupWithForm class